Add jenkinsfile-mode recipe

john2x opened the pull request:
### Brief summary of what the package does

Major mode for editing Jenkins declarative pipeline syntax

### Direct link to the package repository

https://github.com/john2x/jenkinsfile-mode

### Your association with the package

Maintainer

### Relevant communications with the upstream package maintainer

None needed

### Checklist

Please confirm with `x`:

- [x] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses).
- [x] I've read [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [x] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [x] My elisp byte-compiles cleanly
- [x] `M-x checkdoc` is happy with my docstrings
- [x] I've built and installed the package using the instructions in [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [ ] I have confirmed some of these without doing them

riscy commented:
Quick first pass.

### jenkinsfile-mode.el ###

byte-compile (using Emacs 26.3):
```
jenkinsfile-mode.el:30:1:Warning: defcustom for
    `jenkinsfile-mode-vim-source-url' fails to specify type
jenkinsfile-mode.el:30:1:Warning: defcustom for
    `jenkinsfile-mode-vim-source-url' fails to specify type
jenkinsfile-mode.el:35:7:Warning: assignment to free variable
    `jenkinsfile-mode--file-section-keywords'
jenkinsfile-mode.el:37:7:Warning: assignment to free variable
    `jenkinsfile-mode--directive-keywords'
jenkinsfile-mode.el:40:7:Warning: assignment to free variable
    `jenkinsfile-mode--option-keywords'
jenkinsfile-mode.el:46:7:Warning: assignment to free variable
    `jenkinsfile-mode--core-step-keywords'
jenkinsfile-mode.el:52:7:Warning: assignment to free variable
    `jenkinsfile-mode--pipeline-step-keywords'
jenkinsfile-mode.el:225:23:Warning: reference to free variable
    `jenkinsfile-mode--file-section-keywords'
jenkinsfile-mode.el:226:23:Warning: reference to free variable
    `jenkinsfile-mode--directive-keywords'
jenkinsfile-mode.el:227:23:Warning: reference to free variable
    `jenkinsfile-mode--option-keywords'
jenkinsfile-mode.el:228:23:Warning: reference to free variable
    `jenkinsfile-mode--core-step-keywords'
jenkinsfile-mode.el:229:23:Warning: reference to free variable
    `jenkinsfile-mode--pipeline-step-keywords'
jenkinsfile-mode.el:224:7:Warning: assignment to free variable
    `jenkinsfile-mode-font-lock-keywords'
jenkinsfile-mode.el:232:41:Warning: reference to free variable
    `jenkinsfile-mode-font-lock-keywords'
jenkinsfile-mode.el:231:7:Warning: assignment to free variable
    `jenkinsfile-mode-font-lock-defaults'
```
Use `defvar` in these cases.

checkdoc (using version 0.6.1):
- No issues!

package-lint (using version 20200616.2257):
- No issues!

Suggestions/experimental static checks:
- jenkinsfile-mode.el#L19: Prefer `https` over `http` if possible ([why?](https://news.ycombinator.com/item?id=22933774))

### Loadability ###

Verifying ability to #'load each file:
```
Loading jenkinsfile-mode.el
Done.
```

### Package ###

<!-- LICENSE.txt excerpt: `Copyright (C) 2019 John Louis Del Rosario...` -->
- GitHub API found `Other`
  - Try to use a standard format for your license file.
    See: https://github.com/licensee/licensee

john2x commented:
Thanks riscy. Updated to use `defvar`, fixed the `http` usage, and update the license to use the full GPLv3 text.

FYI Re: the `http` usage, it was in a comment. Might be something the experimental check would want to ignore.

riscy commented:
Great, thanks!  I like the automatic poaching from Jenkinsfile.vim :)
